extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpRecorder.java

```diff
@@ -95,22 +117,66 @@ private OtlpGrpcSpanExporter createOtlpGrpcSpanExporter(OtlpExporterRuntimeConfi
                     String[] parts = header.split("=", 2);
                     String key = parts[0].trim();
                     String value = parts[1].trim();
-                    exporterBuilder.addHeader(key, value);
+                    headersMap.put(key, value);
                 }
             }
         }
 
-        if (exporterRuntimeConfig.traces().compression().isPresent()) {
-            exporterBuilder.setCompression(exporterRuntimeConfig.traces().compression().get().getValue());
-        }
+        return new VertxGrpcExporter(
+                "otlp", // use the same as OTel does
+                "span", // use the same as OTel does
+                MeterProvider::noop,
+                ExporterBuilderUtil.validateEndpoint(endpoint),
+                compressionEnabled,
+                tracesConfig.timeout(),
+                headersMap,
+                new Consumer<>() {
+                    @Override
+                    public void accept(HttpClientOptions options) {
+                        configureTLS(options);
+                    }
 
-        if (exporterRuntimeConfig.traces().protocol().isPresent()) {
-            if (!exporterRuntimeConfig.traces().protocol().get().equals(HTTP_PROTOBUF)) {
-                throw new IllegalStateException("Only the GRPC Exporter is currently supported. " +
-                        "Please check `quarkus.otel.exporter.otlp.traces.protocol` property");
-            }
-        }
+                    private void configureTLS(HttpClientOptions options) {
+                        // TODO: this can reuse existing stuff when https://github.com/quarkusio/quarkus/pull/33228 is in
+                        options.setKeyCertOptions(toPemKeyCertOptions(tracesConfig));
+                        options.setPemTrustOptions(toPemTrustOptions(tracesConfig));
+
+                        if (!options.getPemTrustOptions().getCertPaths().isEmpty()
+                                || !options.getPemKeyCertOptions().getCertPaths().isEmpty()
+                                || !options.getPemKeyCertOptions().getKeyPaths().isEmpty()) {
+                            options.setSsl(true);
+                            options.setUseAlpn(true);
```

Comment:
@cescoffier I found that is is necessary, can you confirm?

Reply:
I mean that based on the error message I was seeing, it seemed like Alpn is necessary when using HTTP2. Did I understand correctly?